crates/telio-traversal/src/endpoint_providers/stun.rs

```diff
@@ -328,7 +374,8 @@ enum StunState {
 }
 
 pub struct StunSockets {
-    pub tun_socket: UdpSocket,
+    pub tun_socket_v4: Option<Arc<UdpSocket>>,
+    pub tun_socket_v6: Option<Arc<UdpSocket>>,
```

Comment:
I find this confusing and error prone. We are only using one of those, first ipv6 and if it doesn't work ipv4, right? If that is the case wouldn't it be simpler to have one socket? Start with ipv6 and when we need to switch close it and try ipv4?